File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.py
class Solution(object):
    def kthCharacter(self, k):
        """
        :type k: int
        :rtype: str
        """
        ind = k-1
        inc = 0
        while ind > 0:
            p = 1
            while p * 2 <= ind:
                p *= 2
            inc += 1
            ind -= p
        fin_char = ord('a') + (inc % 26)

        return chr(fin_char)        
        

File: 3600-find-the-k-th-character-in-string-game-i/find-the-k-th-character-in-string-game-i.php
class Solution {

    /**
     * @param Integer $k
     * @return String
     */
    function kthCharacter($k) { 
    $index = $k - 1; 
    $increments = 0;
        
    while ($index > 0) { 
       $a = 1;
        while ($a * 2 <= $index) {
            $a *= 2;
        }
        $increments++; 
        $index -= $a;
    } 
    return chr(ord('a') + ($increments % 26));
    }
}